src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,248 @@
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const source = _.get(attachmentItem, 'source', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        this.props.onNavigate({source: addEncryptedAuthTokenToURL(source), file});
+
+        return {
+            source,
+            file,
+        };
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} shouldShowArrow
+     */
+    toggleArrowsVisibility(shouldShowArrow) {
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
+        let page;
+        const actions = ReportActionsUtils.getSortedReportActions(_.values(this.props.reportActions), true);
+
+        /**
+         * Looping to filter out attachments and retrieve the src URL and name of attachments.
+         */
+        const attachments = [];
+        _.forEach(actions, ({originalMessage, message}) => {
+            // Check for attachment which hasn't been deleted
+            if (!originalMessage || !originalMessage.html || _.some(message, m => m.isEdited)) {
+                return;
+            }
+            const matches = [...originalMessage.html.matchAll(CONST.REGEX.ATTACHMENT_DATA)];
+
+            // matchAll captured both source url and name of the attachment
+            if (matches.length === 2) {
+                const [originalSource, name] = _.map(matches, m => m[2]);
+
+                // Update the image URL so the images can be accessed depending on the config environment.
+                // Eg: while using Ngrok the image path is from an Ngrok URL and not an Expensify URL.
+                const source = tryResolveUrlFromApiRoot(originalSource);
+                if (source === this.state.source) {
+                    page = attachments.length;
+                }
+
+                attachments.push({source, file: {name}});
+            }
+        });
+
+        const {file} = this.getAttachment(attachments[page]);
+        this.setState({
+            page,
+            attachments,
+            file,
+            isForwardDisabled: page === 0,
+            isBackDisabled: page === attachments.length - 1,
+        });
+    }
+
+    /**
+     * Increments or decrements the index to get another selected item
+     * @param {Number} deltaSlide
+    */
+    cycleThroughAttachments(deltaSlide) {
+        if ((deltaSlide < 0 && this.state.isForwardDisabled) || (deltaSlide > 0 && this.state.isBackDisabled)) {
+            return;
+        }
+
+        this.setState(({attachments, page}) => {
+            const nextIndex = page + deltaSlide;
+
+            // Check if index is near the end of the list to fetch more reports
+            if (attachments.length - nextIndex < 10) {
```

Comment:
To test the issue here, use a very chat that has a lot of messages, over 100 or 200.
1. Then upload two attachments. click on one to expand.
2. Click on previous button and you'll notice there is a high chance nothing will happen since it makes a call to load reportActions even though you can see another image outside of the carousel. 

We should fix this usecase since a lot of our DMs already have over 1000 messages and they will grow a lot soon with public channels.

Reply:
@chiragsalian I think I might have solved for this, but it is difficult for me set up a chat like that. If you have any accounts with chats like that I'd be happy to try it out though.

Reply:
i think the updates are not perfect. Okay take this for example,
1. Say a report has 1000 comments. 
2. Assume only the first comment and the last comment are attachments.
3. Now when you load the chat and see the most recent 50 actions and see one attachment, once you click on the attachment and click prev, loading 50 actions at a time to get to the first action that contains a comment seems very inefficient to me.

Let me know your thoughts. We can also see what others suggest incase I'm missing something obvious.
To be clear i think your solution works fine to show attachments when the actions are already loaded in. I have concerns it may not work well for chats containing thousand of actions which is very easily possible.

Reply:
@chiragsalian I think the simplest solution is to query the attachments specifically from the actions in the backend with an offset. I think if there's a flag I could set to just get back the attachments then that would be best too.

Reply:
I agree that [this](https://github.com/Expensify/App/pull/9279#discussion_r1117711559) seems like the best solution, but I think that we should tackle that in a follow up issue, given that we'll have backend work as well.

Reply:
Yeah okay, as I think about this more I think a follow-up for this feature would be good. If we do pursue [this](https://github.com/Expensify/App/pull/9279#discussion_r1117711559) as a follow-up issue I'm wondering if we should tweak the feature here to never call ReadOldestAction i.e., in this PR we only show left/right arrows based on the attachments that are already loaded in onyx? 
So if the user has 50 actions in onyx and in those 50 actions there are 3 images, then when we click on an image we can go next/prev only between these 3 images and never call ReadOldestAction. If the user closes the carousel, scroll up to load more action such that now they have 5 attachments loaded in then clicking an image allows you to navigate between these 5 attachments?

Does that make sense to keep the current feature a little simpler? Because the ReadOldestAction functionality in the current PR code is a bit weird to me.

> I think the simplest solution is to query the attachments specifically from the actions in the backend with an offset. I think if there's a flag I could set to just get back the attachments then that would be best too.

Yup definitely possible. I think for now I'm liking the follow-up idea though.

Reply:
Yeah @chiragsalian, I'm okay with followup. This would need some additional planning discussions with the design too!

cc: @JediWattson